### PR TITLE
Updated dependencies to Dart3/Flutter 3.10 compatible ones

### PR DESCRIPTION
**Pull request by MagnusJohansson**
Update the SDK to v3, additional updates for various packages

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
 name: azure_application_insights
 description: A Dart client to integrate with Azure's Application Insights service
-version: 3.1.0
+version: 3.1.1
```
Comment (kentcb) [inline pubspec.yaml]: Since the SDK major version has incremented, I think we should bump this package to version `4.0.0`?

File: pubspec.yaml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=3.0.0 <4.0.0'
 
 dependencies:
   crypto: ^3.0.0
-  http: ^0.13.0
-  logging: ^1.0.2
-  meta: ^1.3.0
-  stack_trace: ^1.10.0
+  http: '>=0.13.6 <2.0.0'
```
Comment (kentcb) [inline pubspec.yaml:11]: Why the change in syntax here? Can we just do `^1.0.0`?

**Review by kentcb — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some comments

File: pubspec.yaml (hunk 2 above)
Comment (MagnusJohansson) [reply pubspec.yaml:11]: For sure you can go with a single entry `^1.0.0`. But then everybody who are using this package are forced to also use the http package version 1.0.0 - <2.0.0.
If you use `'>=0.13.6 <2.0.0'` , you will be compatible with http version 0.13.6 up to (but not including) 2.0.0.
In other words, both developers who must be on legacy 0.x.x can do so , as well as devs who are using the later 1.x.x

Specifying a single version (with caret):   `^0.13.6` is the same as `'>=0.13.6 <1.0.0'` (i.e. excluding v1.0.0).

Reference:
[Caret syntax](https://dart.dev/tools/pub/dependencies#caret-syntax)
[Versioning](https://dart.dev/tools/pub/versioning)

File: pubspec.yaml (hunk 1 above)
Comment (MagnusJohansson) [reply pubspec.yaml]: Yes, I would say it's a good idea. I've updated the PR.

File: pubspec.yaml (hunk 2 above)
Comment (kentcb) [reply pubspec.yaml:11]: Ah, I get it now. I thought that using SDK 3 would require the use of http 1.0.0 onwards, but I see that it's still possible to use older versions with newer SDKs.

**Comment by kentcb** [comment]
Looks like there are some problems with the build, presumably as a result of SDK 3. I don't have time right now to dig in, but am happy to do so when I free up. If you're keen, feel free to take a look @MagnusJohansson.

**Comment by MagnusJohansson** [comment]
Yes, I see... it seems that the Dart team  in their infinite wisdom decided to mark the[ ByteStream class as final](https://github.com/dart-lang/http/pull/920) (instead of interface) which prevents Mockito to mock it...
I raised an [issue with them](https://github.com/dart-lang/http/issues/956), let's see what their response is.
I suppose a workaround , as suggested in this [Mockito issue](https://github.com/dart-lang/build/issues/4443), is to create a wrapper around http.

**Comment by MagnusJohansson** [comment]
@kentcb Can the tests be modified to not mock the ByteStream, as per this comment?

dart-lang/http#956

**Comment by kentcb** [comment]
Resolved by dart-lang/build#4815 and published
